docs/migration.md

```diff
@@ -1,24 +1,256 @@
 # Migration Notes
 
-## from 2.2 to 3.0
+## From 2.5 to 3.0
 
-This release removes the old, HTTP method-based scopes and replaces them with
-the new, functional-motivated scopes `ADMIN` and `ME`. OSIAM will
-automatically migrate all well-known clients to the new scopes in the following
-way:
+In this release, the auth-server and resource-server have been merged into a single application.
+Therefore you need to merge the databases and configuration files, too.
+Also, the configuration file format is now YAML and deployment has changed.
+This document will guide you through the necessary steps to update OSIAM to version 3.0.
 
-1. Remove the old scopes (`GET`, `POST`, `PUT`, `PATCH`, `DELETE`)
-2. Add the new scopes (`ADMIN`, `ME`)
+Prerequisites:
 
-Well-known clients and their scopes are:
+- You are using the latest version of OSIAM 2.x
+- You are using the latest version of the add-ons (if applicable)
+- Your Java Applications are using at least version 1.8 of the Connector4Java.
 
-* `example-client` (`ADMIN`, `ME`)
-* `addon-self-administration-client` (`ADMIN`)
-* `addon-administration-client` (`ADMIN`)
+**NOTE:** It's highly recommended to try the migration on a staging system, before updating a production system.
+You should thoroughly test your complete staging system.
+And, of course, always make a backup of all the data before updating production!
 
-**NOTE:** The migration will not touch self-defined scopes nor self-defined
-clients. Before deploying OSIAM (and thus run the automatic migration) update
-your existing clients:
+### Update Your JVM
+
+OSIAM now requires at least Java 8.
+Update Java on your machines, if you did not do it already.
+OSIAM runs with both Oracle Java 8 and OpenJDK 8.
+
+### Create OSIAM's Home Directory
+
+OSIAM now loads assets and configuration from a folder in the filesystem.
+Create a home directory on the command line now:
```

Comment:
"Create this directory on the command line now:" you didn't mention that it was the home directory until here. An alternative would be "Create this directory now, we will refer to it as OSIAM's home directory from now on"